spec/features/consumer/registration_spec.rb

```diff
@@ -94,6 +97,43 @@
       expect(e.twitter).to eq "@TwItTeR"
       expect(e.instagram).to eq "@InStAgRaM"
     end
+
+  end
+
+  describe "Terms of Service agreement" do
+    let!(:user2) { create(:user) }
+
+    before do
+      quick_login_as user2
+    end
+
+    context "if accepting Terms of Service is not required" do
+      before { Spree::Config.enterprises_require_tos = false }
+
+      it "allows registration as normal" do
+        visit registration_path
+
+        click_button "Let's get started!"
+        find("div#progress-bar").visible?.should be_true
+      end
+    end
+
+    context "if accepting Terms of Service is required" do
+      before { Spree::Config.enterprises_require_tos = true }
+
+      it "does not allow registration unless checkbox is checked" do
+        visit registration_path
+
+        page.should have_content('Terms of Service')
+        page.should have_selector("input.button.primary[disabled]")
+
```

Comment:
I found a few references online to Spree::Config values responding strangely in tests due to the way they are cached: https://groups.google.com/forum/#!topic/spree-user/93HYDl3ZcPo

Seemed to offer an explanation for the issue, but the suggested fixes didn't help.